### DISCUSSION TITLE
A problem caused by simple flight when flying higher

### DISCUSSION THREAD

**Discussion by cxr001** [microsoft/AirSim#4760] [Support Q&A]
When using movebyVelocity function with simple flight (cascade controller), why the attitude is different when the altitude is different?
I have run the HelloDrone.cpp and use code moveToZAsync and then moveByVelocityAsync to control the blocks drone. If the drone flies to a higher position (eg. z=-100), then use moveByVelocity, the attidude of the drone will be quite strange, usually cannot stay at that position. 

Is this situation normal? Why the attitude is like that? Is there any solution to solve this problem?

ps: If I use moveByRollPitchYawZ, this problem usually would not happen.

**Comment by cxr001** [comment]
I solved it by changing velocity control algorithm

**Reply by mosheFuks** [reply]
Hi  @cxr001 ! Im facing the same issue, how do you solve it?

**Reply by mosheFuks** [reply]
Can you share the solution?
